src/dbally/llm_client/base.py

```diff
@@ -19,7 +22,37 @@ class LLMOptions(ABC):
     Abstract dataclass that represents all available LLM call options.
     """
 
-    dict = asdict
+    _not_given: ClassVar[Optional[LLMOptionsNotGiven]] = None
+
+    def __or__(self, other: "LLMOptions") -> "LLMOptions":
+        """
+        Merges two LLMOptions, prioritizing non-NOT_GIVEN values from the 'other' object.
+        """
+        self_dict = asdict(self)
+        other_dict = asdict(other)
+
+        updated_dict = {
+            key: other_dict.get(key, self_dict[key])
+            if not isinstance(other_dict.get(key), NotGiven)
+            else self_dict[key]
+            for key in self_dict
+        }
+
+        return self.__class__(**updated_dict)
+
+    def dict(self) -> Dict[str, Any]:
+        """
+        Creates a dictionary representation of the LLMOptions instance.
+        If a value is None, it will be replaced with the _not_given value.
```

Comment:
I'm not a fun of mentioning private members in public documentation - I think either it should be public (if it is of interest to the end users) or not mentioned at all. So maybe just "replaced with a provider-specific not-given sentinel"?

Reply:
You're right, thanks for catching

Reply:
You're right, thanks for catching